x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -324,3 +318,76 @@ func TestCalcSingleAssetInAndOut_InverseRelationship(t *testing.T) {
 		}
 	}
 }
+
+func TestCalcJoinPoolShares(t *testing.T) {
```

Comment:
I think these changes are good to be deleted for now since we got most of this testing covered in 
https://github.com/osmosis-labs/osmosis/blob/main/x/gamm/pool-models/balancer/amm_joinpool_test.go